src/status_im/contexts/wallet/sheets/select_account/view.cljs

```diff
@@ -25,12 +28,26 @@
         token                          (->> tokens
                                             (filter #(= (:symbol %) asset-symbol))
                                             first)
-        token-balance                  (get-in token [:balances-per-chain (:chain-id network) :balance])]
+        token-decimals                 (:decimals token)
+        token-balance                  (money/bignumber
+                                        (number/convert-to-whole-number
+                                         (get-in token
+                                                 [:balances-per-chain
+                                                  (:chain-id network) :raw-balance]
+                                                 0)
+                                         token-decimals))
+        display-decimals               (min token-decimals rounding-decimals)
+        token-balance-display          (number/remove-trailing-zeroes (.toFixed token-balance
+                                                                                display-decimals))
+        token-balance-display          (if (and (= token-balance-display "0")
+                                                (money/greater-than token-balance
```

Comment:
all of this belongs in a subscription, cause the component only needs/uses `token-balance-display`. There are similar changes in [this PR](https://github.com/status-im/status-mobile/pull/21116/files), but for the `send` flow.